dispatcher/hub/commonHub.go

```diff
@@ -73,45 +63,30 @@ func checkArgs(args ArgsCommonHub) error {
 	return nil
 }
 
-// Run is launched as a goroutine and listens for events on the exposed channels
-// TODO: use protection from triggering multiple times
-func (ch *commonHub) Run() error {
+// RegisterListener creates a goroutine and listens for WS register events
+func (ch *commonHub) RegisterListener() error {
```

Comment:
why not launch the go routine on constructor?

Reply:
this component is using in 2 ways, firstly to handle websocket subscriptions, and secondly as a handler for pushing block events (which runs in the common publisher goroutine); so I wanted to trigger it separately to be very clear, as for the others in notifierRunner